pkg/apis/apps/validation/validation_test.go

```diff
@@ -463,118 +467,49 @@ func TestValidateStatefulSet(t *testing.T) {
 		},
 		{
 			name: "invalid update strategy",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.OrderedReadyPodManagement,
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: "foo"},
-				},
-			},
+			set:  makeSet(&validLabels, &validPodTemplate, tweakRepicas(3), tweakUpdateStrategy(makeUpdateStrategy(tweakUpdateStrategyType("foo")))),
 			errs: field.ErrorList{
 				field.Invalid(field.NewPath("spec", "updateStrategy"), nil, ""),
 			},
 		},
 		{
 			name: "empty update strategy",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.OrderedReadyPodManagement,
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: ""},
-				},
-			},
+			set:  makeSet(&validLabels, &validPodTemplate, tweakRepicas(3), tweakUpdateStrategy(makeUpdateStrategy(tweakUpdateStrategyType("")))),
 			errs: field.ErrorList{
 				field.Required(field.NewPath("spec", "updateStrategy"), ""),
 			},
 		},
 		{
 			name: "invalid rolling update",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.OrderedReadyPodManagement,
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					UpdateStrategy: apps.StatefulSetUpdateStrategy{Type: apps.OnDeleteStatefulSetStrategyType,
-						RollingUpdate: func() *apps.RollingUpdateStatefulSetStrategy {
-							return &apps.RollingUpdateStatefulSetStrategy{Partition: 1}
-						}()},
-				},
-			},
+			set:  makeSet(&validLabels, &validPodTemplate, tweakRepicas(3), tweakUpdateStrategy(makeUpdateStrategy(tweakUpdateStrategyType(apps.OnDeleteStatefulSetStrategyType), tweakRollingUpdatePartition(1)))),
```

Comment:
You might want to linewrap these cases of multiple tweaks, like:

```
set:  makeSet(&validLabels, &validPodTemplate,
     tweakRepicas(3), 
tweakUpdateStrategy(
    makeUpdateStrategy(
        tweakUpdateStrategyType(apps.OnDeleteStatefulSetStrategyType),
        tweakRollingUpdatePartition(1)
))),
```

or something - the goal is to call out what makes this case unique

Reply:
modified as suggested